sdk/python/kfp/dsl/_container_op.py

```diff
@@ -249,6 +170,132 @@ def add_env_variable(self, env_variable):
     self.env_variables.append(env_variable)
     return self
 
+
+class SideCar(ContainerBase):
```

Comment:
SideCar basically is just a k8s Container. You don't really need to specialize it to a type. Can you move all the fields in it to the Container base type?

Reply:
Actually, sidecar in argo is slightly different from k8s container. You still need to keep the class but move the arguments and command to base class. Could you also add mirrorVolumeMounts field to sidecar, which is the only field which is addition to container spec?

Reply:
Sure. Is there somewhere I can see the Argo spec for sidecar? I tried looking at the docs, but it is so sparse - just the example.

Hence, I have no idea the full spec for sidecar.

Reply:
nvm. I found it. I will try to implement the more common/important ones.

https://github.com/argoproj/argo/blob/master/api/openapi-spec/swagger.json